core/primitives/src/types.rs

```diff
@@ -1079,6 +1077,8 @@ pub enum ValidatorKickoutReason {
     },
     /// Enough stake but is not chosen because of seat limits.
     DidNotGetASeat,
+    /// Validator didn't produce enough chunk endorsements.
+    NotEnoughChunkEndorsements { produced: NumBlocks, expected: NumBlocks },
```

Comment:
So all new variants must be appended at the end for backward compatibility? :)

Reply:
Just wondering if there's a good way to prevent these issues in the future.. comments are one thing but they aren't checks

Reply:
That's the simplest thing to do. Alternatively we can bump db version implement automatic db migration from `LegacyValidatorKickoutReason` to new one.

I was scared for a moment because of this enum being exposed in RPC requests in `ValidatorKickoutView`, but now I see that they use `serde`, which index enums by enum type name, not position, so they are safe.

Yeah, we could mark all data types stored in DB so that if engineer changes them, they are warned that type must be either versioned or properly migrated.